retriever/lib/scripts.py

```diff
@@ -1,36 +1,71 @@
+from __future__ import print_function
+
+from future import standard_library
+
+standard_library.install_aliases()
 import csv
 import imp
 import io
 import os
 import sys
-from os.path import join, isfile, getmtime, exists
+import urllib.request
+import urllib.parse
+import urllib.error
+from os.path import join, isfile, getmtime, exists, abspath
 
 from pkg_resources import parse_version
 
-from retriever.lib.compile import compile_json
-from retriever.lib.defaults import SCRIPT_SEARCH_PATHS, VERSION, ENCODING
+from retriever.lib.defaults import SCRIPT_SEARCH_PATHS, VERSION, ENCODING, SCRIPT_WRITE_PATH
+from retriever.lib.load_json import read_json
+
+
+def check_retriever_minimum_version(module):
+    mod_ver = module.retriever_minimum_version
+    m = module.name
+
+    if hasattr(module, "retriever_minimum_version"):
+        if not parse_version(VERSION) >= parse_version("{}".format(mod_ver)):
+            print("{} is supported by Retriever version ""{}".format(m, mod_ver))
+            print("Current version is {}".format(VERSION))
+            return False
+    return True
 
 
 def MODULE_LIST(force_compile=False):
     """Load scripts from scripts directory and return list of modules."""
     modules = []
     loaded_scripts = []
+    if not os.path.isdir(SCRIPT_WRITE_PATH):
+        os.makedirs(SCRIPT_WRITE_PATH)
 
     for search_path in [search_path for search_path in SCRIPT_SEARCH_PATHS if exists(search_path)]:
-        to_compile = [
+        data_packages = [
             file for file in os.listdir(search_path) if file[-5:] == ".json" and
             file[0] != "_" and (
                 (not isfile(join(search_path, file[:-5] + '.py'))) or (
                     isfile(join(search_path, file[:-5] + '.py')) and (
                         getmtime(join(search_path, file[:-5] + '.py')) < getmtime(
                             join(search_path, file)))) or force_compile)]
```

Comment:
Seems like we can drop most of this pile of conditionals now. I don't think we need either of the `isfile` checks or the `getmtime` check or the `force_compile`. I don't think we're using a leading underscore for anything either, so this whole thing could probably become:

```
data_packages = [file for file in os.listdir(seach_path) if file.endswith(".json")]
```